ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -235,6 +238,63 @@ object EventStorageBackend {
   )
 }
 
+trait DataSourceStorageBackend {
+  def createDataSource(
+      jdbcUrl: String,
+      dataSourceConfig: DataSourceStorageBackend.DataSourceConfig =
+        DataSourceStorageBackend.DataSourceConfig(),
+      connectionInitHook: Option[Connection => Unit] = None,
+  )(implicit loggingContext: LoggingContext): DataSource
+}
+
+object DataSourceStorageBackend {
+  case class DataSourceConfig(
+      pgSynchronousCommit: Option[PgSynchronousCommitValue] = None
+  )
+
+  sealed abstract class PgSynchronousCommitValue(val pgSqlName: String)
```

Comment:
Is the Pg prefix justified? Should this be generic instead?

Reply:
Same was raised by @meiersi-da as well: these are db-specific configurations, so I was opting for having ultimate level of configurability here. For example we do not have async commit  for H2, and we have it (currently not configurable) for Oracle, but with entirely different knobs, with entirely different meanings.
Abstracting over Async commit looses precision, and the idea here to have this loss of precision above the StorageBackend interface, to have free hands later on to implement mapping on-demand

Reply:
Is the idea to have other DB-specific flags as fields of the DataSourceConfig?
```
case class DataSourceConfig(
      pgSynchronousCommit: Option[PgSynchronousCommitValue] = None
      oracleSomeConfigKnob: Option[OracleSomeConfigKnobValue] = None
     etc...
  )
```

Reply:
exactly

Reply:
I would suggest to wait until we have async commit really coming for Oracle (if ever).

What I see in the code is that the DbType.AsyncCommitMode is already very much specialized for postgres async commit (so it is not abstracted away, only these configurations are not implemented for none of the backends except postgres). It is on, off, and local. (Local is expected to be needed by canton) Just for contrast in oracle we have two flags: immediate/batch, and wait/no wait. For me it is not trivial we can draw a one-size-fits-all abstraction over these. But this we will only know, if we will have the oracle usage. Until then I think it should be alright to take a postgre-style async commit config DbType.AsyncCommitMode and mapping this to a postgres specific configuration for the backend.

This concern was raised multiple times, so I would like to emphasize this here again: this is exposed on the Interface because this is exactly what we wish to configure, and only for one type of database-type. This is kind of an exception from the rule here. I was asking myself the question: what if we want to configure db specific datasource properties? Doing this with jdbc-rul properties? Doing this by passing untyped properties down to the driver? Or making them explicit and type safe and expose them on the interface? I was opting for the latter. Happy to continue this discussion :eyes:

Reply:
@nmarton-da : your rationale makes sense to me.

IMHO, the code would be totally fine with the Pg-specific option-only *provided* that option is annotated with comment containing the rationale you give here. I have a hunch that such a comment would even have sufficed to not have a discussion on this topic on the PR at all.

@mziolekda @rautenrieth-da what do you think?

Reply:
Yes, let's add a code comment and be done with the subject

Reply:
Good idea with gaining more clarity this-regard in code @meiersi-da , I added some comments, and also restructured the DataSourceConfig, so it reflects this particular nature: configurations regarding specific StorageBackend implementations.
Also moved the postgres specific part into the backend.postgresql package for more clarity.